### ISSUE TITLE
PMD Tool Fails Locally but CI Doesn't Detect It #18479

### ISSUE THREAD

**Issue by Pankraz76** [checkstyle/checkstyle#18470]
### Summary
Running PMD locally fails, but the CI pipeline doesn't catch this failure. The issue appears after splitting tools in the CI setup.

### Details
After splitting the tools in our CI configuration, we've encountered two issues that question the overall CI setup.

### Related Links
- https://github.com/checkstyle/checkstyle/pull/18440
  - https://github.com/checkstyle/checkstyle/pull/18440/changes#r2655335514

### Expected Behavior
1. PMD should run successfully both locally and in CI
2. CI should fail when PMD encounters issues
3. Local and CI behavior should be consistent

### Current Behavior
1. PMD fails when running locally
2. CI pipeline continues without detecting the PMD failure
3. Inconsistent behavior between local and CI environments

### Impact
- Developers may miss critical code quality issues
- Inconsistent feedback between local development and CI
- Potential quality issues may reach production

### Notes
Need to investigate why CI isn't catching the PMD failures that occur locally. This suggests a configuration discrepancy between local and CI environments.

```
[INFO] 
[INFO] --- pmd:3.28.0:check (default-cli) @ checkstyle ---
[WARNING] PMD Failure: com.puppycrawl.tools.checkstyle.AbstractAutomaticBean:243 Rule:ExceptionAsFlowControl Priority:3 Exception thrown at line 237 is caught in this block..
[WARNING] PMD Failure: com.puppycrawl.tools.checkstyle.DetailNodeTreeStringPrinter:80 Rule:UnusedPrivateMethod Priority:3 Avoid unused private methods such as 'parseJavadocAsDetailNode(String)'..
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:17 min
[INFO] Finished at: 2025-12-31T11:27:50Z
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-pmd-plugin:3.28.0:check (default-cli) on project checkstyle: PMD 7.20.0 has found 2 violations. For more details see: /home/circleci/project/target/pmd.xml -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-pmd-plugin:3.28.0:check (default-cli) on project checkstyle: PMD 7.20.0 has found 2 violations. For more details see: /home/circleci/project/target/pmd.xml
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:333)
    at org.apache.maven.lifecycle.inte
```

**Comment by Pankraz76** [comment]
check needs no quality the 100% quality goal on the website is just as usually talking shit and doing nothing good


if you dont want to understand of purpose its fine to me, no need to waste time on this silly time waster project.

just wanted to help fix your broken software for over one year not constantly broken and the game continues.

**Comment by Pankraz76** [comment]
Lets first fix CI to see broken code...

**Comment by Pankraz76** [comment]
cmd passing:
due to wrong command!

`./mvnw -e --no-transfer-progress test-compile clean pmd:check`

correct

`./mvnw -e --no-transfer-progress clean pmd:check`


IDEA not:

```
/bin/sh ./mvnw -Didea.version=2025.3.1 -Dmaven.ext.class.path=/Users/vincent.potucek/Library/Application Support/JetBrains/IntelliJIdea2025.3/plugins/maven/lib/maven-event-listener.jar -Djansi.passthrough=true -Dstyle.color=always -Dmaven.repo.local=/Users/vincent.potucek/.m2/repository pmd:check -f pom.xml
Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF-8
WARNING: A restricted method in java.lang.System has been called
WARNING: java.lang.System::load has been called by org.fusesource.jansi.internal.JansiLoader in an unnamed module (file:/Users/vincent.potucek/.m2/wrapper/dists/apache-maven-3.9.6/a53741d1/lib/jansi-2.4.0.jar)
WARNING: Use --enable-native-access=ALL-UNNAMED to avoid a warning for callers in this module
WARNING: Restricted methods will be blocked in a future release unless native access is enabled

WARNING: A terminally deprecated method in sun.misc.Unsafe has been called
WARNING: sun.misc.Unsafe::objectFieldOffset has been called by com.google.common.util.concurrent.AbstractFuture$UnsafeAtomicHelper (file:/Users/vincent.potucek/.m2/wrapper/dists/apache-maven-3.9.6/a53741d1/lib/guava-32.0.1-jre.jar)
WARNING: Please consider reporting this to the maintainers of class com.google.common.util.concurrent.AbstractFuture$UnsafeAtomicHelper
WARNING: sun.misc.Unsafe::objectFieldOffset will be removed in a future release
[INFO] Scanning for projects...
[INFO] Inspecting build with total of 1 modules
[INFO] Installing Central Publishing features
[INFO] 
[INFO] ------------------< com.puppycrawl.tools:checkstyle >-------------------
[INFO] Building checkstyle 13.0.1-SNAPSHOT
[INFO]   from pom.xml
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] >>> pmd:3.28.0:check (default-cli) > :pmd @ checkstyle >>>
[INFO] 
[INFO] --- pmd:3.28.0:pmd (pmd) @ checkstyle ---
[INFO] PMD version: 7.20.0
[INFO] Rendering content with org.apache.maven.skins:maven-fluido-skin:jar:2.0.1 skin
[INFO] 
[INFO] <<< pmd:3.28.0:check (default-cli) < :pmd @ checkstyle <<<
[INFO] 
[INFO] 
[INFO] --- pmd:3.28.0:check (default-cli) @ checkstyle ---
[WARNING] PMD Failure: com.puppycrawl.tools.checkstyle.AbstractAutomaticBean:22 Rule:UnnecessaryImport Priority:4 Unused import 'java.beans.PropertyDescriptor'.
[WARNING] PMD Failure: com.puppycrawl.tools.checkstyle.AbstractAutomaticBean:229 Rule:CollapsibleIfStatements Priority:3 This if statement could be combined with its parent.
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  3.213 s
[INFO] Finished at: 2026-01-06T16:10:56+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-pmd-plugin:3.28.0:check (default-cli) on project checkstyle: PMD 7.20.0 has found 2 violations. For more details see: /Users/vincent.potucek/IdeaProjects/checkstyle/target/pmd.xml -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
```

**Comment by Pankraz76** [comment]
@romani please update issue its proven my dedicated PR that CI is leaking...

**Comment by Pankraz76** [comment]
fixed, special thanks to @adangel and @romani
